### ISSUE TITLE
Rails 5.1 testing action_controller_test_request needs param

### ISSUE THREAD

**Issue by schorsch** [trailblazer/cells-rails#26]
::ActionController::TestRequest.create, now needs the controller as param for .create:

https://github.com/trailblazer/cells-rails/blob/master/lib/cell/rails/testing.rb#L16
needs to changed to something like:

```ruby
if ::Rails.version.start_with?('5.1')
  ::ActionController::TestRequest.create(self.class.controller_class)
elsif ::Rails.version.start_with?('5')
  ::ActionController::TestRequest.create
else
  ::ActionController::TestRequest.new
end
```

**Comment by apotonick** [comment]
Please PR. :heart_eyes:

**Comment by tiagoamaro** [comment]
@apotonick @schorsch thanks for going through this 🙇! PR #27 fixes mini-test, but the given solution still raises errors on `rspec-cells`. Since this is not related to `cells-rails`, IMHO, this issue can be closed :)

**Comment by apotonick** [comment]
How would we fix that in rspec-cells?

**Comment by tiagoamaro** [comment]
@apotonick, I've got around with `rspec-cells` and this current solution, and got the following conclusion:

After debugging `cells-rails`, I saw that the `self.class.controller_class` was exactly the same object as the `Cell::RailsExtensions::Testing#controller_class`'s `controller_class` parameter in the test suite (which uses minitest).

With this assumption, it would only require adding a new parameter to the `#action_controller_test_request` method, which would work for minitest and RSpec:

```ruby
module Cell
  module RailsExtensions
    # This modules overrides Cell::Testing#controller_for and provides Rails-specific logic.
    module Testing
      def controller_for(controller_class)
        return unless controller_class

        controller_class.new.tap do |ctl|
          ctl.request = action_controller_test_request(controller_class)
          ctl.instance_variable_set :@routes, ::Rails.application.routes.url_helpers
        end
      end

      def action_controller_test_request(controller_class)
        if ::Rails::VERSION::MAJOR >= 5
          if ::Rails::VERSION::MINOR >= 1 # ~> 5.1
            ::ActionController::TestRequest.create(controller_class)
          else # 5.0
            ::ActionController::TestRequest.create
          end
        else #< 5
          ::ActionController::TestRequest.new
        end
      end
    end # Testing
  end
end
```
In other words, this would avoid assuming which context `self` call is referring to.

If you agree with this, I'm pretty happy to open a PR with this change :)

**Comment by apotonick** [comment]
Successfully registered gem: cells-rails (0.0.8)
